ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt

```diff
@@ -70,7 +70,7 @@ internal open class ByteBufferChannel(
         job.invokeOnCompletion(onCancelling = true) { cause ->
             attachedJob = null
             if (cause != null) {
-                cancel(CancellationException("Channel closed due to job failure").apply { initCause(cause) })
+                cancel(cause)
```

Comment:
This is potentially dangerous: all write operations everywhere will produce logged errors. Most likely it will undo fixes made before to avoid I/O errors logging. Need to be changed with care

Reply:
Could you please clarify the I/O errors logging issue? I'm not sure I'm aware of it.

Reply:
The problem is that people getting undesired IO exceptions logged at server when working with channels. In real server connection issues are very common including those that caused by browser termination and user cancellation (button stop). At server side users don't want to see such exceptions or want to have them logged at the debug level.

I am not exactly sure, if it leads to such issues. May be not. But for sure this changes behaviour in many places that we can't easily trace in mind.

Reply:
Well, I do understand the problem, but I struggle to understand how the wrapping of all exceptions into `CancellationException` can help to distinguish exceptions that user wants to see in logs from exceptions he doesn't want to see.